pvlib/pvsystem.py

```diff
@@ -944,14 +944,23 @@ def singlediode(self, photocurrent, saturation_current,
                            resistance_series, resistance_shunt, nNsVth,
                            ivcurve_pnts=ivcurve_pnts)
 
-    def i_from_v(self, resistance_shunt, resistance_series, nNsVth, voltage,
-                 saturation_current, photocurrent):
+    def i_from_v(self, voltage, photocurrent, saturation_current,
+                 resistance_series, resistance_shunt, nNsVth):
         """Wrapper around the :py:func:`pvlib.pvsystem.i_from_v` function.
 
-        See :py:func:`pvsystem.i_from_v` for details
+        See :py:func:`pvlib.pvsystem.i_from_v` for details.
+
+        .. versionchanged:: 0.10.0
+           The function's arguments have been reordered. In earlier versions,
+           the argument order is
+
+           .. code-block:: python
+
+              i_from_v(resistance_shunt, resistance_series, nNsVth, voltage,
+                       saturation_current, photocurrent, method='lambertw')
```

Comment:
I don't think including `method` here is correct

Reply:
I included ``method`` here to match how the ``i_from_v`` function declaration is displayed above. Leaving ``method`` out might suggest that ``method`` did not exist in earlier versions. However, the code block makes it look like an example of calling the method, not documentation for a function declaration. Instead of a code block, we can use the `.. py:function::` directive:
Code:
```
.. py:function:: i_from_v(resistance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent, method='lambertw')
```
Render:
![image](https://github.com/pvlib/pvlib-python/assets/41020789/882d9cce-b6b7-41db-bcfe-22ad6491f71f)
The downside is that the ``[source]`` link points to the ``0.10.0`` version of the function which has the updated argument order.

We can also remove the code block. Since we give the version that contains these changes, people can look at the documentation for older releases to see the old argument order.

Reply:
How it looks without the old function declaration:
![image](https://github.com/pvlib/pvlib-python/assets/41020789/dff0da22-d77a-4383-90db-cc8ade65796a)

Reply:
> Leaving `method` out might suggest that method did not exist in earlier versions.

Unless I'm missing something, I think `method` indeed does not exist in `pvlib.pvsystem.PVSystem.i_from_v`.  It only exists in `pvlib.pvsystem.i_from_v` and `pvlib.pvsystem.v_from_i`.

Anyway, I'm fine with the current state of this PR (just saying "The function's arguments have been reordered." and leaving it at that).

Reply:
Ah I see, you are correct. I missed that difference between the two ``i_from_v`` methods.

Sounds good.